shopfloor/actions/search.py

```diff
@@ -14,13 +14,14 @@ class SearchAction(Component):
 
     # TODO: these methods shall be probably replaced by scan anything handlers
 
-    def location_from_scan(self, barcode):
+    def location_from_scan(self, barcode, limit=1):
         model = self.env["stock.location"]
         if not barcode:
             return model.browse()
-        return model.search(
-            ["|", ("barcode", "=", barcode), ("name", "=", barcode)], limit=1
-        )
+        res = model.search([("barcode", "=", barcode)], limit=limit)
+        if len(res) < limit:
+            res += model.search([("name", "=", barcode)], limit=(limit - len(res)))
```

Comment:
I think with += you could return twice the  same location
```suggestion
            res |= model.search([("name", "=", barcode)], limit=(limit - len(res)))
```

Reply:
I don't know about having multiple record in a set ? But better use the one listed in the docs https://www.odoo.com/documentation/14.0/developer/reference/addons/orm.html#operations.
Changed

Reply:
Indeed, the docs talk about using `set` operators on recordsets :wink: 
```
>>> rset1
res.partner(14, 26)
>>> rset2
res.partner(14, 26, 33, 27)
>>> rset1 | rset2
res.partner(14, 26, 33, 27)
>>> rset1 + rset2
res.partner(14, 26, 14, 26, 33, 27)
```
and if you use `+=`
```
>>> rset3
res.partner(14, 26)
>>> rset3 += rset2
>>> rset3
res.partner(14, 26, 14, 26, 33, 27)
```
while if you use `|=`
```
>>> rset3
res.partner(14, 26)
>>> rset3 |= rset2
>>> rset3
res.partner(14, 26, 33, 27)
```